### ISSUE TITLE
Handling of non-relocatable packages is broken in bob v0.19.0

### ISSUE THREAD

**Issue by danielroedl** [BobBuildTool/bob#410]
The [relocatable flag](https://bob-build-tool.readthedocs.io/en/latest/manual/configuration.html?highlight=relocatable#relocatable) is not working correct if it is set to `false` since version 0.19.0.

I provide a test for this here: https://github.com/danielroedl/bob/tree/feature/test_non-relocatable/test/non-relocatable

This test succeed for all bob versions including 0.18.1. With version 0.19.0 it fails.

The first commit the test fails (and maybe the cause) is: https://github.com/BobBuildTool/bob/commit/0d93b3cce3aa5d415b951f20eb66413587020f3f

**Comment by jkloetzke** [comment]
Indeed, the regression is caused by the commit you have found. I've created #411 which should fix the problem and restore the old behaviour. I took the freedom to directly cherry-pick the test case. I hope this is ok.

While fixing I found another problem related to shared packages where basically the same could have happened. This is now fixed too.

**Comment by danielroedl** [comment]
Thanks for the quick response and the fix.
I will test it.

**Comment by danielroedl** [comment]
The test build with #411 was successful.
